### ISSUE TITLE
Windows build please

### ISSUE THREAD

**Issue by jamesmkrieger** [conda-forge/plumed-feedstock#16]
Can we have a Plumed build for Windows please? It would be really nice to be able to use it on Windows laptop where I do most of my software development.

- [ ] I read [the conda-forge documentation](https://conda-forge.org/docs/user/introduction.html#how-can-i-install-packages-from-conda-forge) and could not find the solution for my problem there.

Issue:

<br/>
Environment (<code>conda list</code>):
<details>

```
$ conda list

```
</details>

<br/>
Details about  <code>conda</code> and system ( <code>conda info</code> ):
<details>

```
$ conda info

```
</details>

**Comment by GiovanniBussi** [comment]
@jamesmkrieger thanks for opening this issue. I am not very practical with Windows builds. In general, PLUMED is only supported on unix, but some time ago @tonigi was maintaining a Windows binary. Maybe Toni has some idea on how to make a conda executable for windows? I guess we would then be limited to plumed executable and python wrappers

**Comment by tonigi** [comment]
I used to build windows executables manually:  https://github.com/giorginolab/vmd_plumed/blob/master/doc/PLUMED-WIN32.md (look around for the executables). I used **mingw64** which makes it trivial, they just worked (at least _plumed driver_, which is what I cared about). 

However, this is not the conda-forge way :( Building with MSVC required some changes AND  a way to run the configure bash script under windows. Perhaps it can be done, but I did not dig into it.

**Comment by jamesmkrieger** [comment]
Thanks both of you. I use mingw too so hopefully that'll work.

**Comment by jamesmkrieger** [comment]
It's a shame conda-forge won't do it though. I remember it was a pain to couple plumed to gromacs a few years ago even on linux and the existence of an conda-forge option for plumed with openmm seemed very promising until I realised it doesn't work for windows.

**Comment by GiovanniBussi** [comment]
BTW, if you need gromacs + plumed on conda (linux/osx, no windows) you can install it following the instructions of our masterclasses (https://www.plumed.org/doc-v2.7/user-doc/html/masterclass-21-3.html).

**Comment by tonigi** [comment]
Also, it may be of interest that [condacolab](https://colab.research.google.com/drive/1c_RGCgQeLHVXlF44LyOFjfUW32CmG6BP) may enable people to run the conda packages on Colab right away.

**Comment by jamesmkrieger** [comment]
> 
> 
> BTW, if you need gromacs + plumed on conda (linux/osx, no windows) you can install it following the instructions of our masterclasses (https://www.plumed.org/doc-v2.7/user-doc/html/masterclass-21-3.html).

Much easier than https://groups.google.com/g/plumed-users/c/stlK9-kaa6A/m/WKc5i1xRDgAJ !

**Comment by tonigi** [comment]
Edited to add: the following cells will work nicely to install the gromacs+plumed on colaboratory - see https://pypi.org/project/condacolab/. 

```
!pip install -q condacolab
import condacolab
condacolab.install()
```

execute once, the kernel restarts, then 

```
conda install --strict-channel-priority -c plumed/label/masterclass -c conda-forge gromacs
```

as directed by the masterclass instructions. In no time this should give you an environment where you can run `!gmx`, `!plumed` and so on. 

The only catch, as far as I can tell, is that you won't be able to import Plumed's Python module from inside colab (because it belongs to the conda installation).

**Comment by jamesmkrieger** [comment]
Thanks. Good to know

**Comment by GiovanniBussi** [comment]
A useful suggestion related to windows building on [twitter](https://twitter.com/isuru_f/status/1366137563929575424) (see [this](https://github.com/conda-forge/autotools_clang_conda-feedstock/blob/master/recipe/meta.yaml#L34)).

I'll try to have a look ASAP

**Comment by jamesmkrieger** [comment]
Thanks

**Comment by GiovanniBussi** [comment]
@tonigi related to this, I found out that if you install modules with pip *before* running `condacolab.install()` the modules can be seen from the notebook. Se [here](https://colab.research.google.com/drive/1yrhRL08beBYXVXT-hIEqq8qGVpZfW_Zo?usp=sharing). Cool! Maybe we can publish the solutions to the masterclass also in this manner. Thanks!

> Edited to add: the following cells will work nicely to install the gromacs+plumed on colaboratory - see https://pypi.org/project/condacolab/.
> 
> ```
> !pip install -q condacolab
> import condacolab
> condacolab.install()
> ```
> 
> execute once, the kernel restarts, then
> 
> ```
> conda install --strict-channel-priority -c plumed/label/masterclass -c conda-forge gromacs
> ```
> 
> as directed by the masterclass instructions. In no time this should give you an environment where you can run `!gmx`, `!plumed` and so on.
> 
> The only catch, as far as I can tell, is that you won't be able to import Plumed's Python module from inside colab (because it belongs to the conda installation).

**Comment by tonigi** [comment]
More and more off-topic, it's worthwhile noting that [WSL](https://en.wikipedia.org/wiki/Windows_Subsystem_for_Linux) works well IMHO. One may even argue that it makes building scientific packages for Windows a chore of the past.

**Comment by jamesmkrieger** [comment]
> 
> 
> More and more off-topic, it's worthwhile noting that [WSL](https://en.wikipedia.org/wiki/Windows_Subsystem_for_Linux) works well IMHO. One may even argue that, if it gets traction, it could make building scientific packages for Windows a chore of the past. (Although not perfect for classes because it takes a bit to install)

Yes, I agree. I am trying WSL now. I also found it was an easier way to get OpenMPI and Java, which I need for some other programs too.

**Comment by sukritsingh** [comment]
Hi – just wanted to bump this and ask if there's been any progress in getting PLUMED support for Windows onto conda-forge? Or is this not on the roadmap anytime soon? I'd like to get a Windows version running on a few machines, but totally understand if it's not on the roadmap anymore.

**Comment by GiovanniBussi** [comment]
@sukritsingh you can see that state here: https://github.com/conda-forge/plumed-feedstock/pull/17

Basically, I was stuck. If anyone wants to help I would be happy to check it again and eventually merge it.

**Comment by sukritsingh** [comment]
Awesome! Thanks for the info!

I am not experienced with Windows builds or the feedstock but my labmate Mike Henry (@mikemhenry), a  Software Scientist in our group, has volunteered his time to help out!

**Comment by mikemhenry** [comment]
Working on this here: https://github.com/conda-forge/plumed-feedstock/pull/32

It looks like `gawk` isn't available for windows, so I will have to get that build first here https://github.com/conda-forge/gawk-feedstock/pull/18

**Comment by tonigi** [comment]
My attempts are here if useful https://github.com/conda-forge/plumed-feedstock/pull/31
